File: IronPython/WCF/sample1/test_server.py

import clr
clr.AddReference("System.ServiceModel")
clr.AddReference("CommonLib.dll")

from System import *
from System.ServiceModel import *
from CommonLib import *

def copyfile(file):
	print "called copyfile\n"
	return "returned : %s" % file

service = DownloadManager()
#service.CopyFileCallBack = CopyFileDelegate(copyfile)
service.CopyFileCallBack = copyfile

host = ServiceHost(service)

binding = NetNamedPipeBinding(NetNamedPipeSecurityMode.None)

host.AddServiceEndpoint(IDownloadManager, binding, "net.pipe://localhost/test")

host.Open()

Console.WriteLine("press enter key to terminate")
Console.ReadLine()

host.Close()

File: csharp/wcf/test1/TestServer.cs

using System;
using System.ServiceModel;
using CommonLib;

public class TestServer
{
	static void Main(string[] args)
	{
		string uri = "net.pipe://localhost/test";

        DownloadManager dm = new DownloadManager();
        dm.CopyFileCallBack = delegate(string file)
        {
            return string.Format("get : {0}", file);
        };

		using (ServiceHost host = new ServiceHost(dm))
		{
            NetNamedPipeBinding binding = new NetNamedPipeBinding(NetNamedPipeSecurityMode.None);
            host.AddServiceEndpoint(typeof(IDownloadManager), binding, uri);

            host.Open();

			Console.WriteLine("press enter key to terminate");
			Console.ReadLine();

			host.Close();
		}
	}
}
